### DISCUSSION TITLE
Error make Edge from Bspline

### DISCUSSION THREAD

**Discussion by oppakun** [donalffons/opencascade.js#149] [Q&A] [answered]
I try to create bspline from points and make edge and got binding error
```js
let geomCurveHandle = new oc.GeomAPI_PointsToBSpline_2(ptList,3,8,openCascade.GeomAbs_Shape.GeomAbs_C2,0.001).Curve();
let edge = new oc.BRepBuilderAPI_MakeEdge_24(geomCurveHandle).Edge();
```
binding error
`"Expected null or instance of Handle_Geom_Curve, got an instance of Handle_Geom_BSplineCurve"`

**Comment by donalffons** [comment, marked as answer]
Without having tried that myself, but maybe this works?
```js
let edge = new oc.BRepBuilderAPI_MakeEdge_24(new oc.Handle_Geom_Curve_2(geomCurveHandle.get())).Edge();
```

It's impossible to allow all implicit conversions implemented in OCCT from JavaScript, so sometimes you have to be very explicit in your JS code.

**Reply by oppakun** [reply]
It worked. I thought about conversions but I don't know how to do it.
Thank you for your guided ❤️
